scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java

```diff
@@ -644,79 +580,66 @@ public Double doInTransaction(Session session) {
+            Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
+            if (jobQuery.uniqueResult() == null) {
+                return null;
             }
 
+            Query query = session.getNamedQuery("getMeanTaskRunningTime").setParameter("id", id);
+
+            Double result1 = (Double) query.uniqueResult();
+            return result1 == null ? 0 : result1;
         });
 
         return checkResult(id, result);
     }
 
     public int getTotalNumberOfHostsUsed(String jobId) {
         final long id = Long.parseLong(jobId);
-        int result = executeReadOnlyTransaction(new SessionWork<Integer>() {
-            @Override
-            public Integer doInTransaction(Session session) {
-                Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
-                if (jobQuery.uniqueResult() == null) {
-                    return null;
-                }
-
-                Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
-
-                return ((Long) query.uniqueResult()).intValue();
+        int result = executeReadOnlyTransaction(session -> {
+            Query jobQuery = session.getNamedQuery("checkJobExistence").setParameter("id", id);
+            if (jobQuery.uniqueResult() == null) {
+                return null;
             }
 
+            Query query = session.getNamedQuery("getTotalNumberOfHostsUsed").setParameter("id", id);
+
+            return ((Long) query.uniqueResult()).intValue();
         });
 
         return checkResult(id, result);
     }
 
     public SchedulerAccount readAccount(final String username) {
-        return executeReadOnlyTransaction(new SessionWork<SchedulerAccount>() {
-
-            @Override
-            public SchedulerAccount doInTransaction(Session session) {
-                Query tasksQuery = session.getNamedQuery("readAccountTasks").setParameter("username", username);
+        return executeReadOnlyTransaction(session -> {
+            Query tasksQuery = session.getNamedQuery("readAccountTasks").setParameter("username", username);
 
-                int taskCount;
-                long taskDuration;
-
-                Object[] taskResult = (Object[]) tasksQuery.uniqueResult();
-                taskCount = ((Number) taskResult[0]).intValue();
-                if (taskResult[1] != null) {
-                    taskDuration = ((Number) taskResult[1]).longValue();
-                } else {
-                    taskDuration = 0;
-                }
+            int taskCount;
+            long taskDuration;
 
-                int jobCount;
-                long jobDuration;
+            Object[] taskResult = (Object[]) tasksQuery.uniqueResult();
+            taskCount = ((Number) taskResult[0]).intValue();
+            if (taskResult[1] != null) {
+                taskDuration = ((Number) taskResult[1]).longValue();
+            } else {
+                taskDuration = 0;
```

Comment:
Why zero here, it was initialized with zero on top. You could do that explicit on top and remove this else code path.